riscv/insns/mret.h

```diff
@@ -11,8 +11,8 @@ s = set_field(s, MSTATUS_MPP, p->extension_enabled('U') ? PRV_U : PRV_M);
 s = set_field(s, MSTATUS_MPV, 0);
 if (ZICFILP_xLPE(prev_virt, prev_prv)) {
   STATE.elp = static_cast<elp_t>(get_field(s, MSTATUS_MPELP));
-  s = set_field(s, MSTATUS_MPELP, elp_t::NO_LP_EXPECTED);
 }
+s = set_field(s, MSTATUS_MPELP, elp_t::NO_LP_EXPECTED);
```

Comment:
Should have a guard of `p->extension_enabled(EXT_ZICFILP)`

Reply:
I don't think that guard is needed, because the bit is hardwired to 0 when Zicfilp is not implemented, so the statement has no effect.